sklearn/neighbors/base.py

```diff
@@ -44,7 +44,8 @@
 
 VALID_METRICS_SPARSE = dict(ball_tree=[],
                             kd_tree=[],
-                            brute=PAIRWISE_DISTANCE_FUNCTIONS.keys())
+                            brute=[x for x in PAIRWISE_DISTANCE_FUNCTIONS.keys()
```

Comment:
I think this metadata (which metrics don't support sparse) should be in metrics.pairwise, not here

Reply:
ok, moved it to this file and created a dedicated list for clarity sake